MekHQ/src/mekhq/campaign/market/ContractMarket.java

```diff
@@ -182,7 +189,7 @@ public void generateContractOffers(Campaign campaign, boolean newCampaign) {
                         inBackwater = false;
                     }
                 }
-            } else {
+            } else if (currentFactions.size() > 0) {
```

Comment:
Why > 0 instead of == 0?

Reply:
This case is for at least one faction. The above case is for more than one.

Reply:
Sorry, the latter should have been == 1

Reply:
I noticed the static analysis tools we use have problems with ranges mixed with (in)equality and did much better with ranges alone. I believe it was elsewhere with target rolls.